frontend/src/scenes/session-recordings/player/sessionRecordingDataLogic.ts

```diff
@@ -207,10 +208,72 @@ function coerceToEventWithTime(d: unknown, withMobileTransformer: boolean): even
         : (currentEvent as eventWithTime)
 }
 
+function chunkMutationSnapshot(
+    snapshot: RecordingSnapshot,
+    nextSnapshot: RecordingSnapshot | null
+): RecordingSnapshot[] {
+    if (
+        snapshot.type !== EventType.IncrementalSnapshot ||
+        !('data' in snapshot) ||
+        !snapshot.data ||
+        typeof snapshot.data !== 'object' ||
+        !('source' in snapshot.data) ||
+        snapshot.data.source !== IncrementalSource.Mutation ||
+        !('adds' in snapshot.data) ||
+        !Array.isArray(snapshot.data.adds) ||
+        snapshot.data.adds.length <= MUTATION_CHUNK_SIZE
+    ) {
+        return [snapshot]
+    }
+
+    const chunks: RecordingSnapshot[] = []
+    const { adds, removes, texts, attributes } = snapshot.data
+    const totalAdds = adds.length
+    const chunksCount = Math.ceil(totalAdds / MUTATION_CHUNK_SIZE)
+
+    // Calculate the time gap to the next snapshot
+    // If there's no next snapshot, use a small increment (1ms) per chunk
+    const timeGap = nextSnapshot ? nextSnapshot.timestamp - snapshot.timestamp : chunksCount
+    const timeIncrement = timeGap / chunksCount
+
+    // Calculate the delay gap if both snapshots have delays
+    const delayGap = (nextSnapshot?.delay ?? (snapshot.delay ?? 0) + chunksCount) - (snapshot.delay ?? 0)
+    const delayIncrement = delayGap / chunksCount
+
+    for (let i = 0; i < chunksCount; i++) {
+        const startIdx = i * MUTATION_CHUNK_SIZE
+        const endIdx = Math.min((i + 1) * MUTATION_CHUNK_SIZE, totalAdds)
+        const isFirstChunk = i === 0
+        const isLastChunk = i === chunksCount - 1
+
+        const chunkSnapshot: RecordingSnapshot = {
+            ...snapshot,
+            timestamp: Math.floor(snapshot.timestamp + i * timeIncrement),
```

Comment:
so this can just be timestamp + 1 unless it would make it equal to next timestamp, we only want to make sure the order of operations is maintained